statefun-flink/statefun-flink-io-bundle/src/main/java/org/apache/flink/statefun/flink/io/kinesis/AwsAuthConfig.java

```diff
@@ -0,0 +1,128 @@
+      properties.setProperty(AWSConfigConstants.AWS_REGION, awsRegion.asId().id());
+    } else if (awsRegion.isCustomEndpoint()) {
+      final AwsRegion.CustomEndpointAwsRegion customEndpoint = awsRegion.asCustomEndpoint();
+      setCustomEndpointForConsumer(properties, customEndpoint);
+      setCustomEndpointForProducer(properties, customEndpoint);
+    } else {
+      throw new IllegalStateException("Unrecognized AWS region configuration type: " + awsRegion);
+    }
+  }
+
+  private static void setAwsCredentialProperties(
+      Properties properties, AwsCredentials awsCredentials) {
+    if (awsCredentials.isDefault()) {
+      properties.setProperty(
+          AWSConfigConstants.AWS_CREDENTIALS_PROVIDER,
+          AWSConfigConstants.CredentialProvider.AUTO.name());
+    } else if (awsCredentials.isBasic()) {
+      properties.setProperty(
+          AWSConfigConstants.AWS_CREDENTIALS_PROVIDER,
+          AWSConfigConstants.CredentialProvider.BASIC.name());
+
+      final AwsCredentials.BasicAwsCredentials basicCredentials = awsCredentials.asBasic();
+      properties.setProperty(
+          AWSConfigConstants.accessKeyId(AWSConfigConstants.AWS_CREDENTIALS_PROVIDER),
+          basicCredentials.accessKeyId());
+      properties.setProperty(
+          AWSConfigConstants.secretKey(AWSConfigConstants.AWS_CREDENTIALS_PROVIDER),
+          basicCredentials.secretAccessKey());
+    } else if (awsCredentials.isProfile()) {
+      properties.setProperty(
+          AWSConfigConstants.AWS_CREDENTIALS_PROVIDER,
+          AWSConfigConstants.CredentialProvider.PROFILE.name());
+
+      final AwsCredentials.ProfileAwsCredentials profileCredentials = awsCredentials.asProfile();
+      properties.setProperty(
+          AWSConfigConstants.profileName(AWSConfigConstants.AWS_CREDENTIALS_PROVIDER),
+          profileCredentials.name());
+      profileCredentials
+          .path()
+          .ifPresent(
+              path ->
+                  properties.setProperty(
+                      AWSConfigConstants.profilePath(AWSConfigConstants.AWS_CREDENTIALS_PROVIDER),
+                      path));
+    } else {
+      throw new IllegalStateException(
+          "Unrecognized AWS credentials configuration type: " + awsCredentials);
+    }
+  }
+
+  private static String regionFromDefaultProviderChain() {
+    return new DefaultAwsRegionProviderChain().getRegion().toLowerCase(Locale.ENGLISH);
+  }
+
+  private static void setCustomEndpointForConsumer(
+      Properties properties, AwsRegion.CustomEndpointAwsRegion customEndpoint) {
+    properties.setProperty(AWSConfigConstants.AWS_ENDPOINT, customEndpoint.serviceEndpoint());
+    properties.setProperty(AWSConfigConstants.AWS_REGION, customEndpoint.regionId());
+  }
+
+  private static void setCustomEndpointForProducer(
+      Properties properties, AwsRegion.CustomEndpointAwsRegion customEndpoint) {
+    URL url;
+    try {
+      url = new URL(customEndpoint.serviceEndpoint());
```

Comment:
You may want to switch to a URI
If you are only interested in the host.
URL requires a DNS lookup.